src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs

```diff
@@ -1193,20 +1231,36 @@ protected override void EndProcessing()
             {
                 if (ParameterSetName.Equals(InvokeCommandCommand.InProcParameterSet))
                 {
-                    if (_steppablePipeline != null)
+                    Version currentStrictModeVersion = Context.EngineSessionState.CurrentScope.StrictModeVersion;
+                    try
                     {
-                        _steppablePipeline.End();
+                        if (MyInvocation.BoundParameters.ContainsKey(nameof(StrictModeVersion)))
```

Comment:
Sorry for not being clear.  We can remove this setting `Context.EngineSessionState.CurrentScope.StrictModeVersion` here because it is already set in the `BeginProcessing()` method above.